### PR TITLE
enh(OCP\SpeechToText): Introduce ISpeechToTextProviderWithUserId

### PR DESCRIPTION
**Pull request by andrey18106**
## Summary

To keep consistency between SpeechToText, TextProcessing, Translation and TextToImage providers introduce `IProviderWithUserId`.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/private/SpeechToText/SpeechToTextManager.php [hunk 1]
```diff
@@ -123,6 +126,9 @@ public function transcribeFile(File $file): string {
 				if ($provider instanceof ISpeechToTextProviderWithId) {
 					return $provider->getId() === $classNameOrId;
 				}
+				if ($provider instanceof ISpeechToTextProviderWithUserId) {
+					$provider->setUserId($this->userSession->getUser()?->getUID());
+				}
```
Comment (marcelklehr) [inline lib/private/SpeechToText/SpeechToTextManager.php:131]: I think the userId should be set below before the call to transcribeFile, no?

Comment (andrey18106) [reply lib/private/SpeechToText/SpeechToTextManager.php]: You are right, moved it before provider execution

**Review by bigcat88 — APPROVED** [review]

**Review by marcelklehr — APPROVED** [review]

**Comment by nickvergessen** [comment]
CI is fixed by #43543

**Comment by bigcat88** [comment]
*documentation already present for this*

**Comment by nickvergessen** [comment]
Please document in https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/speech-to-text.html and a short summary in https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.html#id1

**Comment by nickvergessen** [comment]
> documentation already present for this

I see the PR, but not the change on the live page.... maybe building broke?

**Comment by nickvergessen** [comment]
At least in https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.html#id1 it should still be added with a link to the normal docs

**Comment by nickvergessen** [comment]
> I see the PR, but not the change on the live page.... maybe building broke?

Ah, merged 1h ago :D

**Pull request merged**